src/modules/hip/kernel/lut.hpp

```diff
@@ -0,0 +1,250 @@
+    dst_f8->f4[1] = make_float4((float)lut[(int) src_f8->f1[4]], (float)lut[(int) src_f8->f1[5]], (float)lut[(int) src_f8->f1[6]], (float)lut[(int) src_f8->f1[7]]);
+}
+
+__device__ void lut_hip_compute(uchar *srcPtr, d_float8 *src_f8, d_float8 *dst_f8, signed char *lut)
+{
+    dst_f8->f4[0] = make_float4((float)lut[(int) src_f8->f1[0]], (float)lut[(int) src_f8->f1[1]], (float)lut[(int) src_f8->f1[2]], (float)lut[(int) src_f8->f1[3]]);
+    dst_f8->f4[1] = make_float4((float)lut[(int) src_f8->f1[4]], (float)lut[(int) src_f8->f1[5]], (float)lut[(int) src_f8->f1[6]], (float)lut[(int) src_f8->f1[7]]);
+}
+
+__device__ void lut_hip_compute(uchar *srcPtr, d_float8 *src_f8, d_float8 *dst_f8, float *lut)
+{
+    dst_f8->f4[0] = make_float4(lut[(int) src_f8->f1[0]], lut[(int) src_f8->f1[1]], lut[(int) src_f8->f1[2]], lut[(int) src_f8->f1[3]]);
+    dst_f8->f4[1] = make_float4(lut[(int) src_f8->f1[4]], lut[(int) src_f8->f1[5]], lut[(int) src_f8->f1[6]], lut[(int) src_f8->f1[7]]);
+}
+
+__device__ void lut_hip_compute(uchar *srcPtr, d_float8 *src_f8, d_float8 *dst_f8, half *lut)
+{
+    dst_f8->f4[0] = make_float4(lut[(int) src_f8->f1[0]], lut[(int) src_f8->f1[1]], lut[(int) src_f8->f1[2]], lut[(int) src_f8->f1[3]]);
+    dst_f8->f4[1] = make_float4(lut[(int) src_f8->f1[4]], lut[(int) src_f8->f1[5]], lut[(int) src_f8->f1[6]], lut[(int) src_f8->f1[7]]);
+}
+
+__device__ void lut_hip_compute(signed char *srcPtr, d_float8 *src_f8, d_float8 *dst_f8, signed char *lut)
+{
+    d_float8 srcNorm_f8;
+    srcNorm_f8.f4[0] = src_f8->f4[0] + (float4)128;
+    srcNorm_f8.f4[1] = src_f8->f4[1] + (float4)128;
+    dst_f8->f4[0] = make_float4((float)lut[(int) srcNorm_f8.f1[0]], (float)lut[(int) srcNorm_f8.f1[1]], (float)lut[(int) srcNorm_f8.f1[2]], (float)lut[(int) srcNorm_f8.f1[3]]);
+    dst_f8->f4[1] = make_float4((float)lut[(int) srcNorm_f8.f1[4]], (float)lut[(int) srcNorm_f8.f1[5]], (float)lut[(int) srcNorm_f8.f1[6]], (float)lut[(int) srcNorm_f8.f1[7]]);
+}
+
+template <typename T1, typename T2>
+__global__ void lut_pkd_tensor(T1 *srcPtr,
+                               uint2 srcStridesNH,
+                               T2 *dstPtr,
+                               uint2 dstStridesNH,
+                               T2 *lutPtr,
+                               RpptROIPtr roiTensorPtrSrc)
+{
+    int id_x = (hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x) * 8;
+    int id_y = hipBlockIdx_y * hipBlockDim_y + hipThreadIdx_y;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    if ((id_y >= roiTensorPtrSrc[id_z].xywhROI.roiHeight) || (id_x >= roiTensorPtrSrc[id_z].xywhROI.roiWidth * 3))
+    {
+        return;
+    }
+
+    uint srcIdx = (id_z * srcStridesNH.x) + ((id_y + roiTensorPtrSrc[id_z].xywhROI.xy.y) * srcStridesNH.y) + (id_x + roiTensorPtrSrc[id_z].xywhROI.xy.x * 3);
+    uint dstIdx = (id_z * dstStridesNH.x) + (id_y * dstStridesNH.y) + id_x;
+
+    d_float8 src_f8, dst_f8;
+
+    rpp_hip_load8_and_unpack_to_float8(srcPtr + srcIdx, &src_f8);
```

Comment:
the kernel is using computations assuming the data_type of Src and Dst whereas the types are templated.
How are we planning to support other data_types for  Src and Dst. Same true for other kernel implementations here

Reply:
The rpp_hip_load8_and_unpack_to_float8() function actually takes any datatype and converts to float.
Also, as we discussed lut is only useful in U8/I8 inputs, but the outputs can be U8/I8/F16/F32 for which we have support.
A [256] length or [65536] length look up table could be provided by the user.